### PR TITLE
ffi: add fd_bmtree32 and fd_bmtree20 to bindings

### PR DESCRIPTION
**Pull request by anoushk1234**

### CODE REVIEW

**Comment by mmcgee-jump** [comment]
Please rebase on latest main, and I think `cargo fmt` will place one import on each line. Otherwise looks good.

**Comment by anoushk1234** [comment]
> Please rebase on latest main, and I think `cargo fmt` will place one import on each line. Otherwise looks good.

@mmcgee-jump have made the changes!

File: ffi/rust/firedancer-sys/src/ballet/mod.rs [hunk 1]
```diff
@@ -4,10 +4,12 @@ mod sha256;
...
 mod txn;
+mod bmtree;
 
 pub use ed25519::*;
 pub use sbpf::*;
 pub use sha256::*;
 pub use sha512::*;
 pub use shred::*;
 pub use txn::*;
+pub use bmtree::*;
```
Comment (mmcgee-jump) [inline ffi/rust/firedancer-sys/src/ballet/mod.rs:15]: Can you also add a newline at the end here :)

Comment (anoushk1234) [reply ffi/rust/firedancer-sys/src/ballet/mod.rs]: > Can you also add a newline at the end here :)

yep done!

**Comment by mmcgee-jump** [comment]
Last thing can you squash the 3 commits :)

**Comment by anoushk1234** [comment]
@mmcgee-jump yep done

**Comment by ptaffet-jump** [comment]
Just a heads up that bmtree got refactored pretty heavily in b681acfbbc338d7a7719a4b209c46ca86c231210. That commit hasn't made it to main yet, but will at some point.

**Comment by anoushk1234** [comment]
> Just a heads up that bmtree got refactored pretty heavily in [b681acf](https://github.com/firedancer-io/firedancer/commit/b681acfbbc338d7a7719a4b209c46ca86c231210). That commit hasn't made it to main yet, but will at some point.

yes @ptaffet-jump that's fine by us.

**Comment by mmcgee-jump** [comment]
Looks like there's still a rustfmt issue in CI ^

**Comment by anoushk1234** [comment]
> Looks like there's still a rustfmt issue in CI ^

looks like fmt is formatting it horizontally instead, for this reason I had to manually format it.
image

**Comment by mmcgee-jump** [comment]
Maybe your rustfmt settings are overriding the ones in the project? You can see the errors it produces by clicking into the rustfmt CI error above,

```/home/runner/.cargo/bin/cargo fmt --all -- --check```

```Diff in /home/runner/work/firedancer/firedancer/ffi/rust/firedancer-sys/src/ballet/bmtree.rs at line 1:
 pub use crate::generated::{
-    fd_bmtree20_commit, 
-    fd_bmtree20_commit_append, 
+    fd_bmtree20_commit,
+    fd_bmtree20_commit_append,
     fd_bmtree20_commit_fini,
-    fd_bmtree20_commit_init, 
-    fd_bmtree20_commit_leaf_cnt, 
+    fd_bmtree20_commit_init,
+    fd_bmtree20_commit_leaf_cnt,
     fd_bmtree20_commit_t,
-    fd_bmtree20_hash_leaf, 
-    fd_bmtree20_node, 
-    fd_bmtree20_node_t, 
+    fd_bmtree20_hash_leaf,
+    fd_bmtree20_node,
+    fd_bmtree20_node_t,
     fd_bmtree32_commit,
-    fd_bmtree32_commit_append, 
-    fd_bmtree32_commit_fini, 
+    fd_bmtree32_commit_append,
+    fd_bmtree32_commit_fini,
     fd_bmtree32_commit_init,
-    fd_bmtree32_commit_leaf_cnt, 
-    fd_bmtree32_commit_t, 
-    fd_bmtree32_hash_leaf, 
+    fd_bmtree32_commit_leaf_cnt,
+    fd_bmtree32_commit_t,
+    fd_bmtree32_hash_leaf,
     fd_bmtree32_node,
     fd_bmtree32_node_t,
 };
Diff in /home/runner/work/firedancer/firedancer/ffi/rust/firedancer-sys/src/ballet/mod.rs at line 1:
+mod bmtree;
 mod ed2551[9](https://github.com/firedancer-io/firedancer/actions/runs/5741611711/job/15564208558?pr=557#step:4:10);
 mod sbpf;
 mod sha256;
Diff in /home/runner/work/firedancer/firedancer/ffi/rust/firedancer-sys/src/ballet/mod.rs at line 4:
 mod sha5[12](https://github.com/firedancer-io/firedancer/actions/runs/5741611711/job/15564208558?pr=557#step:4:13);
 mod shred;
 mod txn;
-mod bmtree;
 
+pub use bmtree::*;
 pub use ed255[19](https://github.com/firedancer-io/firedancer/actions/runs/5741611711/job/15564208558?pr=557#step:4:20)::*;
 pub use sbpf::*;
 pub use sha[25](https://github.com/firedancer-io/firedancer/actions/runs/5741611711/job/15564208558?pr=557#step:4:26)6::*;
Diff in /home/runner/work/firedancer/firedancer/ffi/rust/firedancer-sys/src/ballet/mod.rs at line 12:
 pub use sha[51](https://github.com/firedancer-io/firedancer/actions/runs/5741611711/job/15564208558?pr=557#step:4:52)2::*;
 pub use shred::*;
 pub use txn::*;
-pub use bmtree::*;
 
Error: The process '/home/runner/.cargo/bin/cargo' failed with exit code 1```

**Comment by mmcgee-jump** [comment]
Sorry for all the back and forth about this trivial CI stuff. You can see the command it's running and make sure it works locally as well, also can you keep the change squashed :)

```cargo fmt --all -- --check```

**Comment by anoushk1234** [comment]
> cargo fmt --all -- --check

yes I've used this but the issue is the checks don't make use of the rustfmt config from the project, I've tried different things but can't seem to get it to use that config. I will squash it once I confirm CI doesn't fail.

**Comment by anoushk1234** [comment]
@mmcgee-jump not sure why CI is not running on latest commit but that should have fixed the fmt issues as per the previous CI logs. Can you please check it out?

**Comment by anoushk1234** [comment]
Let me squash and try again

**Comment by anoushk1234** [comment]
@mmcgee-jump looks like the CI requires approval, can you please approve it?

**Review by mmcgee-jump — APPROVED** [review]

**Comment by ripatel-fd** [comment]
Thank you for the contribution 🙏

**Pull request merged**